### DISCUSSION TITLE
Use image-set from next/image in bg class

### DISCUSSION THREAD

**Discussion by florianbepunkt** [tailwindlabs/tailwindcss#14943] [Help] [answered]
Instead of 

```
<div class="bg-[url('/img/hero-pattern.svg')]">
```

I would like to use the image-set coming from nextjs:

```
import { getImageProps } from 'next/image'
 
function getBackgroundImage(srcSet = '') {
  const imageSet = srcSet
    .split(', ')
    .map((str) => {
      const [url, dpi] = str.split(' ')
      return `url("${url}") ${dpi}`
    })
    .join(', ')
  return `image-set(${imageSet})`
}

return (
<div className={`bg-[${getBackgroundImage(srcSetLandscape)}]`} />
)
```

However this does not seem to work. Any help would be appreciated.

**Comment by wongjn** [comment]
As per [the documentation](https://tailwindcss.com/docs/content-configuration#dynamic-class-names):

> The most important implication of how Tailwind extracts class names is that it will only find classes that exist as _complete unbroken strings_ in your source files.
>
> If you use string interpolation or concatenate partial class names together, Tailwind will not find them and therefore will not generate the corresponding CSS:
>
> **Don’t construct class names dynamically**
>
> ```vue
> <div class="text-{{ error ? 'red' : 'green' }}-600"></div>
> ```
> In the example above, the strings `text-red-600` and `text-green-600` do not exist, so Tailwind will not generate those classes.
> Instead, make sure any class names you’re using exist in full:
>
> **Always use complete class names**
>
> ```vue
> <div class="{{ error ? 'text-red-600' : 'text-green-600' }}"></div>
> ``` 

Instead, you could consider using the `style` attribute like:

```diff
 }

 return (
-<div className={`bg-[${getBackgroundImage(srcSetLandscape)}]`} />
+  <div style={{ backgroundImage: getBackgroundImage(srcSetLandscape) }} />
 )
```

**Reply by florianbepunkt** [reply]
Thank you, but using the style attribute is not the answer, as I need to change the bg image based on the screen size.

**Reply by wongjn** [reply, marked as answer]
You can still use the `style` attribute with variants, like:

```jsx
<div
  class="bg-[image:var(--image-1)] md:bg-[image:var(--image-2)]"
  style={{
    '--image-1: image1,
    '--image-2: image2,
  }}
/>
```

**Reply by florianbepunkt** [reply]
Awesome. Thank you!
